packages/docusaurus-theme-common/src/utils/usePluralForm.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {useMemo} from 'react';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+
+type LocalePluralForms = {
+  locale: string;
+  pluralForms: Intl.LDMLPluralRule[];
+  select: (count: number) => Intl.LDMLPluralRule;
+};
+
+// Hardcoded english/fallback implementation
+const EnglishPluralForms: LocalePluralForms = {
+  locale: 'en',
+  pluralForms: ['one', 'other'],
+  select: (count) => (count === 1 ? 'one' : 'other'),
+};
+
+function createLocalePluralForms(locale: string): LocalePluralForms {
+  const pluralRules = new Intl.PluralRules(locale);
+  return {
+    locale,
+    pluralForms: pluralRules.resolvedOptions().pluralCategories,
+    select: (count) => pluralRules.select(count),
+  };
+}
+
+// Poor man's PluralSelector implementation, using an english fallback.
+// We want a lightweight, future-proof and good-enough solution.
+// We don't want a perfect and heavy solution.
+//
+// Docusaurus classic theme has only 2 deeply nested labels requiring complex plural rules
+// We don't want to use Intl + PluralRules polyfills + full ICU syntax (react-intl) just for that.
+//
+// Notes:
+// - 2021: 92+% Browsers support Intl.PluralRules, and support will increase in the future
+// - NodeJS >= 13 has full ICU support by default
+// - In case of "mismatch" between SSR and Browser ICU support, React keeps working!
+function useLocalePluralForms(): LocalePluralForms {
+  const {
+    i18n: {currentLocale},
+  } = useDocusaurusContext();
+  return useMemo(() => {
+    if (Intl && Intl.PluralRules) {
+      try {
+        return createLocalePluralForms(currentLocale);
+      } catch (e) {
+        console.error(`Failed to use Intl.PluralRules for locale=${currentLocale}.
+Docusaurus will fallback to a default/fallback (English) Intl.PluralRules implementation.
+`);
+        return EnglishPluralForms;
+      }
+    } else {
+      console.error(`Intl.PluralRules not available!
+Docusaurus will fallback to a default/fallback (English) Intl.PluralRules implementation.
+        `);
+      return EnglishPluralForms;
+    }
+  }, [currentLocale]);
+}
+
+function selectPluralMessage(
+  pluralMessages: string,
+  count: number,
+  localePluralForms: LocalePluralForms,
+): string {
+  const separator = '|';
```

Comment:
u need to handle escape & potentially double escape here as well (e.g what if I want to use the `|` as a literal?)

Reply:
This utility is not a public API and will only be used by 2 labels in the theme. 
I don't think it is too important as none of those labels should contain "|", and the 2 labels we need to pluralize will be provided by default out of the box.

But that probably wouldn't harm either to escape, will take a look

Reply:
ok makes sense :) ignore me then. I thought this is gonna be a public syntax